src/rt-app.c

```diff
@@ -970,6 +970,35 @@ static void set_thread_priority(thread_data_t *data, sched_data_t *sched_data)
 			exit(EXIT_FAILURE);
 	}
 
+	if (sched_data->policy != deadline &&
+	    (sched_data->util_min != -1 || sched_data->util_max != -1)) {
+		sa_params.size = sizeof(struct sched_attr);
+		sa_params.sched_flags = SCHED_FLAG_KEEP_ALL;
+		tid = gettid();
+
+		if (sched_data->util_min != -1) {
+			sa_params.sched_util_min = sched_data->util_min;
+			sa_params.sched_flags |= SCHED_FLAG_UTIL_CLAMP_MIN;
+			log_notice("[%d] setting util_min=%d",
+				   data->ind, sched_data->util_min);
+		}
+		if (sched_data->util_max != -1) {
+			sa_params.sched_util_max = sched_data->util_max;
+			sa_params.sched_flags |= SCHED_FLAG_UTIL_CLAMP_MAX;
+			log_notice("[%d] setting util_max=%d",
+				   data->ind, sched_data->util_max);
+		}
+
+		ret = sched_setattr(tid, &sa_params, flags);
+		if (ret != 0) {
+			log_critical("[%d] sched_setattr returned %d",
+				     data->ind, ret);
+			errno = ret;
```

Comment:
Sorry I missed this earlier. From the man pages sched_setattr() will return -1 and set errno. I can see that other places in the code have similar logic. Am I missing something or we're effectively setting errno to -1 here?

Reply:
No, that's wrong... lemme get rid of that assignment, as you correctly do in #83.